cibuildwheel/options.py

```diff
@@ -594,37 +620,122 @@ def check_for_deprecated_options(self) -> None:
-            for option_name, option_value in sorted(dataclasses.asdict(self.globals).items())
-        ]
+        lines = []
+        global_option_names = sorted(f.name for f in dataclasses.fields(self.globals))
 
-        build_option_defaults = self.build_options(identifier=None)
+        for option_name in global_option_names:
+            option_value = getattr(self.globals, option_name)
+            default_value = getattr(self.defaults.globals, option_name)
+            lines.append(self.option_summary(option_name, option_value, default_value))
+
+        build_options = self.build_options(identifier=None)
+        build_options_defaults = self.defaults.build_options(identifier=None)
         build_options_for_identifier = {
             identifier: self.build_options(identifier) for identifier in identifiers
         }
 
-        for option_name, default_value in sorted(dataclasses.asdict(build_option_defaults).items()):
+        build_option_names = sorted(f.name for f in dataclasses.fields(build_options))
+
+        for option_name in build_option_names:
             if option_name == "globals":
                 continue
 
-            lines.append(f"{option_name}: {default_value!r}")
+            option_value = getattr(build_options, option_name)
+            default_value = getattr(build_options_defaults, option_name)
+            overrides = {
+                i: getattr(build_options_for_identifier[i], option_name) for i in identifiers
+            }
 
-            # if any identifiers have an overridden value, print that too
-            for identifier in identifiers:
-                option_value = getattr(build_options_for_identifier[identifier], option_name)
-                if option_value != default_value:
-                    lines.append(f"  {identifier}: {option_value!r}")
+            lines.append(
+                self.option_summary(option_name, option_value, default_value, overrides=overrides)
+            )
 
         return "\n".join(lines)
 
+    def option_summary(
+        self,
+        option_name: str,
+        option_value: Any,
+        default_value: Any,
+        overrides: dict[str, Any] | None = None,
+    ) -> str:
+        """
+        Return a summary of the option value, including any overrides, with
+        ANSI 'dim' color if it's the default.
+        """
+        value_str = self.option_summary_value(option_value)
+        default_value_str = self.option_summary_value(default_value)
+        overrides_value_strs = {
+            k: self.option_summary_value(v) for k, v in (overrides or {}).items()
+        }
+        # if the override value is the same as the non-overridden value, don't print it
+        overrides_value_strs = {k: v for k, v in overrides_value_strs.items() if v != value_str}
+
+        has_been_set = (value_str != default_value_str) or overrides_value_strs
+        c = log.colors
+
+        result = ""
+        result += c.gray if not has_been_set else ""
```

Comment:
```suggestion
        result = c.gray if not has_been_set else ""
```

? Otherwise, I'm not sure this is a great use of a ternary, I'd probably just waste one more line and use a normal if.

Reply:
Yep 👍  this might just be a remnant from some refactoring